pkg/client/add_events.go

```diff
@@ -227,28 +229,18 @@ func (client *DataSetClient) Shutdown() error {
 		RandomizationFactor: client.Config.BufferSettings.RetryRandomizationFactor,
 		Multiplier:          client.Config.BufferSettings.RetryMultiplier,
 		MaxInterval:         client.Config.BufferSettings.RetryMaxInterval,
-		MaxElapsedTime:      client.Config.BufferSettings.RetryMaxElapsedTime,
+		MaxElapsedTime:      client.Config.BufferSettings.RetryShutdownTimeout / 2,
```

Comment:
If we want to say in the documentation - "set grace period to the same value" - then I would say, that we want to finish in something like 90% of the time, to be sure. There is also some additional work done - so I can imagine this value to be around 40%?


What do you think?

Reply:
Do we need to be that accurate? I am aware that there might be some imprecision but i suppose it is good enough.